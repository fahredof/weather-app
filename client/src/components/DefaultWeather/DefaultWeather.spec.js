import React from 'react';
import DefaultWeather from "./DefaultWeather";

describe(`DefaultWeather`, () => {
    const mainCity = {
        city: `Ufa`,
        image: `https://openweathermap.org/img/wn/04n@2x.png`,
        temp: '-11'
    };
    const mainCityError = {
        city: ``,
        image: ``,
        temp: ''
    };
    const props = {
        mainCity
    };

    describe(`отображение города`, () => {
        const defCity = shallow(<DefaultWeather {...props}/>);
        it(`отображается корректно`, () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });

    describe('ошибка при получении города', () => {
        const nextProps = {
            ...props,
            mainCity: mainCityError
        };
        const defCity = shallow(<DefaultWeather {...nextProps} />);
        it('отображается корректно', () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });
});