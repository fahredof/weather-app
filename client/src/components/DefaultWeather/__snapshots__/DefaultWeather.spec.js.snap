// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultWeather загрузка отображается корректно 1`] = `
<div
  className="def"
  id="def"
>
  <div
    className="default"
  >
    <p
      className="city"
    >
      Ufa
    </p>
    <p
      className="temp"
    >
      -11
    </p>
    <div
      className="image"
    >
      <img
        alt=""
        src="https://openweathermap.org/img/wn/04n@2x.png"
      />
    </div>
  </div>
</div>
`;

exports[`DefaultWeather отображение города отображается корректно 1`] = `
<div
  className="def"
  id="def"
>
  <div
    className="default"
  >
    <p
      className="city"
    >
      Ufa
    </p>
    <p
      className="temp"
    >
      -11
    </p>
    <div
      className="image"
    >
      <img
        alt=""
        src="https://openweathermap.org/img/wn/04n@2x.png"
      />
    </div>
  </div>
</div>
`;

exports[`DefaultWeather ошибка при получении города отображается корректно 1`] = `
<div
  className="def"
  id="def"
>
  <Spinner />
</div>
`;

exports[`DefaultWeather ошибка при получении города ошибка при получении города 1`] = `null`;
